src/main/scala/dd2d/DdmToDataverseMapper.scala

```diff
@@ -49,22 +52,19 @@ class DdmToDataverseMapper() {
     val RD_SRS_NAME = "http://www.opengis.net/def/crs/EPSG/0/28992"
   }
 
-  def mapFilesToJson(filesXml: Node): Seq[FileMetadata] = {
-    val files = new ListBuffer[FileMetadata]
+  def extractFileInfoFromFilesXml(filesXml: Node): Seq[FileInformation] = {
+    val files = new ListBuffer[FileInformation]
     (filesXml \\ "file").filter(_.nonEmpty).foreach(n => {
       val description = (n \ "description").headOption.map(_.text)
       val directoryLabel = (n \ "@filepath").headOption.map(_.text)
       //todo how to map permissions?
       val restrict = Some("false")
-      files += FileMetadata(description, directoryLabel, restrict)
+      files += FileInformation(File(projectRootToFilePathAttribute + directoryLabel.get),
```

Comment:
Convention: prefix options with `maybe`.
Extract val to prevent complex nested arguments.
`get` is not safe, create a meaningful message, or don't create a `FileInformation`.
```scala
      val maybeDescription = (n \ "description").headOption.map(_.text)
      val maybeFilePath = (n \ "@filepath").headOption.map(_.text)
      val maybeParent = maybeFilePath.map(File(_).parent.pathAsString)
      val file = maybeFilePath.map(projectRootToFilePathAttribute / _)
        .getOrElse(throw new IllegalArgumetException("..."))```